tests/test_configset_in.py

```diff
@@ -33,7 +33,7 @@ def test_configset_from_ConfigSets(tmp_path):
     assert len([at for at in c_ats]) == 4
 
 
-@pytest.mark.xfail()
+@pytest.mark.xfail(reason="should fail creating ConfigSet_in from mix of ConfigSet_ins using input files and input configs")
```

Comment:
Why are you using xfail? My understanding is that `xfail` is for tests that fail because the code is incomplete or yet to be fixed and the cases the code is expected to fail should be handled via `with pytest.raises(RuntimeError): ...`. It seems to me that this is the latter case?

Reply:
Yes - I think I misunderstood xfail.  I'll update that.  We should also check to make sure xfail isn't being used incorrectly in other places, I suppose.

Reply:
OK, fixed now.  The only other use of xfail is in the QE test, and that's the way it's intended.